### PR TITLE
enhance: Use batch to speed up list collections from meta kv

### PR DESCRIPTION
**Pull request by weiliu1031**
issue: #36228

### CODE REVIEW

**Comment by weiliu1031** [comment]
rerun go-sdk

File: internal/metastore/kv/rootcoord/kv_catalog.go [hunk 1]
```diff
@@ -677,23 +796,28 @@ func (kc *Catalog) ListCollections(ctx context.Context, dbID int64, ts typeutil.
...
-			return nil, err
-		}
-		colls = append(colls, collection)
+		kc.fixDefaultDBIDConsistency(ctx, collMeta, ts)
+		colls = append(colls, collMeta)
+	}
+	log.Info("unmarshal all collection details cost", zap.Int64("db", dbID), zap.Duration("cost", time.Since(start)))
+
+	start = time.Now()
+	ret, err := kc.batchAppendPartitionAndFieldsInfo(ctx, colls, ts)
```
Comment (jaime0815) [inline internal/metastore/kv/rootcoord/kv_catalog.go:814]: Does this mean we must list all partitions of all collections even if only one collection is requested?

**Comment by xiaofan-luan** [comment]
1. I think we can remove the the naming xx210
2. func (kc *Catalog) batchListPartitionsAfter210(ctx context.Context, ts typeutil.Timestamp) (map[int64][]*model.Partition, error) {
	_, values, err := kc.Snapshot.LoadWithPrefix(PartitionMetaPrefix, ts)
	if err != nil {
		return nil, err
	}

	ret := make(map[int64][]*model.Partition)
	for i := 0; i < len(values); i++ {
		partitionMeta := &pb.PartitionInfo{}
		err := proto.Unmarshal([]byte(values[i]), partitionMeta)
		if err != nil {
			return nil, err
		}
		collectionID := partitionMeta.GetCollectionId()
		if ret[collectionID] == nil {
			ret[collectionID] = make([]*model.Partition, 0)
		}
		ret[collectionID] = append(ret[collectionID], model.UnmarshalPartitionModel(partitionMeta))
	}
	return ret, nil
}

this might be a lot.

For example, if we get 1000 collections * 1000 partitions, there will be 1m meta fetched at one time.
Maybe we should put it into iterator mode.
But anyway I believe since should be much faster when collection is number is huge.

**Pull request merged**